ompi/mca/coll/libnbc/nbc_ireduce.c

```diff
@@ -656,6 +656,8 @@ static inline int red_sched_redscat_gather(
     /* Find nearest power-of-two less than or equal to comm_size */
     int nsteps = opal_hibit(comm_size, comm->c_cube_dim + 1);   /* ilog2(comm_size) */
     int nprocs_pof2 = 1 << nsteps;                              /* flp2(comm_size) */
+    if (nprocs_pof2 < 2)
+        return OMPI_ERR_NOT_SUPPORTED;
```

Comment:
Super minor nit: please always use `{}`, even for 1-line blocks.

And a minor question: will returning non-SUCCESS here cause a problem farther up the stack?  I.e., will it cause the overall operation to think that it has failed, and propagate that failure all the way back up to the MPI API?  It might be worth putting a short comment here about this case (e.g., if this case can/should never happen, because this function should not normally be called with the wrong number of procs, ...etc.).

Reply:
@jsquyres I added a comment to the commit. This case (nprocs_pof2 < 2) never happens and do not cause a problem further up the stack. In the case of comm_size < 2 other algorithms for MPI_Ireduce are used. This check is only for Coverity.